python/ray/train/v2/jax/jax_trainer.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from typing import TYPE_CHECKING, Callable, Dict, Optional, Union
+
+from ray.air._internal.config import ensure_only_allowed_dataclass_keys_updated
+from ray.train import Checkpoint, DataConfig
+from ray.train.trainer import GenDataset
+from ray.train.v2.api.config import RunConfig, ScalingConfig
+from ray.train.v2.api.data_parallel_trainer import DataParallelTrainer
+from ray.train.v2.jax.config import JaxConfig
+from ray.util import PublicAPI
+
+if TYPE_CHECKING:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+@PublicAPI(stability="alpha")
+class JaxTrainer(DataParallelTrainer):
```

Comment:
We'll want to add this to the [API references](https://github.com/ray-project/ray/blob/master/doc/source/train/api/api.rst) but if the dependencies aren't working well we can fast-follow.